core/modules/qmeta/QMeta.h

```diff
@@ -272,6 +273,37 @@ class QMeta {
     virtual std::vector<QueryId> getQueriesForTable(std::string const& dbName,
                                                     std::string const& tableName) = 0;
 
+    /**
+     * @brief Create the table for temporary query statistics.
+     * @throw SqlError
+     */
+    virtual void createQueryStatsTmpTable() = 0;
+
+    /**
+     * @brief Insert a row for tracking chunksCompleted vs totalChunks of a query.
+     * @return true if successful.
+     */
+    virtual bool queryStatsTmpRegister(QueryId queryId, int totalChunks) = 0;
+
+    /**
+     * @brief Update the number of completed chunks
+     * @return true if successful.
+     */
+    virtual bool queryStatsTmpChunkUpdate(QueryId queryId, int completedChunks) = 0;
+
+    /**
+     * @brief Get statistics for queryId
+     * @return QStats object containing query completion information.
+     * @throw QueryIdError, SqlError
+     */
+    virtual QStats queryStatsTmpGet(QueryId queryId) = 0;
+
+    /**
+     * @ brief Remove row for completion status when the query is done.
+     * @return true if successful.
+     */
+    virtual bool queryStatsTmpRemove(QueryId queryId) = 0;
```

Comment:
I had thought this information was going to be put in the central database in in-memory tables?

Reply:
My impression was that we wanted to avoid pounding central server with the updates that no one reads (or reads very rarely). Still, if it's going to be a central service I think some extra flexibility may be beneficial, we could move it to some specialized in memory-service like memcached or whatnot.